internal/pkg/v2/application/usecases/common/version/version.go

```diff
@@ -0,0 +1,43 @@
+/*******************************************************************************
+ * Copyright 2020 Dell Inc.
+ *
+ * Licensed under the Apache License, UseCase 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+// version implements behavior for the version use case.
+package version
+
+import (
+	"github.com/edgexfoundry/edgex-go/internal/pkg/v2/application"
+	dtoBase "github.com/edgexfoundry/edgex-go/internal/pkg/v2/application/dto/v2dot0/base"
+	dto "github.com/edgexfoundry/edgex-go/internal/pkg/v2/application/dto/v2dot0/version"
+)
+
+// UseCase contains references to dependencies required by the corresponding Routable contract implementation.
+type UseCase struct{}
+
+// New is a factory method that returns an initialized UseCase receiver struct.
+func New() *UseCase {
+	return &UseCase{}
+}
+
+// Execute encapsulates the behavior for this use case.
+func (uc *UseCase) Execute(r interface{}) (interface{}, application.Status) {
+	request, _ := r.(*dto.Request)
+	return dto.NewResponse(dtoBase.NewResponseForSuccess(request.RequestID), application.VersionLatest),
```

Comment:
The version endpoint is suppose to return the service's release version, i.e. 1.1.0 or when this released 2.0.0, not the API version.

Reply:
The specification for VersionResponse.version states "[the latest version supported by this service](https://github.com/edgexfoundry/edgex-go/blob/d8a62da57cabe0c8b37a1721a5b06be65ae47022/api/openapi/v2/core-command.yaml#L266)."   

This is ambiguous -- you think service's release version and I think latest version of the API supported by the service.

There is value in having both.  Do you think I should extend the OpenAPI VersionResponse definition to include both?

Reply:
I think this needs to be discussing in Core WG. The App SDK uses the V1 version to validate that the core service release version is compatible with the SDK version. i.e., the major values match.

Reply:
@tsconn23 note request to discuss in working group above.

To clarify, I'm suggesting we return -- as separate properties -- both the service's version number (which for Geneva will be 1.x) and the latest major.minor version of the supported API (which for Geneva will be 2.0) in the APIv2 version endpoint.  No change would be made to the data returned by the existing V1 endpoint.

Yes, this is a breaking change.  But so is the entirety of APIv2.

Reply:
Works for me.

Reply:
(unresolved to keep on my radar for implementation.)

Reply:
Discussed in today's Core WG call.

Version endpoint is not versioned in URL.

Will update v2 implementation to match existing v1 implementation.  Will remove v2's version endpoint to remove ../v2/.. from URI path.  Will take v2 version endpoint out of common routing wire-up for Geneva.

Reply:
On further investigation, I'm not sure the above-mentioned approach will work well.  I'm going to leave the /api/v2/version endpoint in place.  I intend to defer decision on when/how to implement /api/version for after the Geneva release.